omnigibson/prims/entity_prim.py

```diff
@@ -994,86 +995,110 @@ def get_relative_linear_velocity(self):
+
+        Returns:
+            2-tuple:
+                - th.Tensor: (x,y,z) position in the specified frame
+                - th.Tensor: (x,y,z,w) quaternion orientation in the specified frame
+        """
+
+        assert frame in ["world", "parent", "scene"], f"Invalid frame '{frame}'. Must be 'world', 'parent', or 'scene'."
+
         # If the simulation isn't running, we should read from this prim's XForm (object-level) properties directly
         if og.sim.is_stopped():
-            return XFormPrim.get_position_orientation(self, clone=clone)
+            return XFormPrim.get_position_orientation(self, frame=frame, clone=clone)
         # Delegate to RigidPrim if we are not articulated
         elif self._articulation_view is None:
-            return self.root_link.get_position_orientation(clone=clone)
+            return self.root_link.get_position_orientation(frame=frame, clone=clone)
         # Sim is running and articulation view exists, so use that physx API backend
         else:
-            positions, orientations = self._articulation_view.get_world_poses(clone=clone)
-            return positions[0], orientations[0][[1, 2, 3, 0]]
+            if frame == "world" or frame == "scene":
+                positions, orientations = self._articulation_view.get_world_poses(clone=clone)
+            else:
+                positions, orientations = self._articulation_view.get_local_poses()
 
-    def set_local_pose(self, position=None, orientation=None):
-        # If kinematic only, clear cache for the root link
-        if self.kinematic_only:
-            self.root_link.clear_kinematic_only_cache()
-        # If the simulation isn't running, we should set this prim's XForm (object-level) properties directly
-        if og.sim.is_stopped():
-            return XFormPrim.set_local_pose(self, position, orientation)
-        # Delegate to RigidPrim if we are not articulated
-        elif self._articulation_view is None:
-            self.root_link.set_local_pose(position=position, orientation=orientation)
-        # Sim is running and articulation view exists, so use that physx API backend
-        else:
-            if position is not None:
-                position = th.asarray(position)[None, :]
-            if orientation is not None:
-                orientation = th.asarray(orientation)[None, [3, 0, 1, 2]]
-            self._articulation_view.set_local_poses(position, orientation)
-            PoseAPI.invalidate()
+        position, orientation = positions[0], orientations[0][[1, 2, 3, 0]]
+
+        # If we are in a scene, compute the scene-local transform
+        if frame == "scene":
+            if self.scene is None:
+                raise ValueError("Cannot transform position and orientation relative to scene without a scene")
+            else:
+                position, orientation = T.mat2pose(self.scene.pose_inv @ T.pose2mat((position, orientation)))
+
+        return position, orientation
```

Comment:
Do the same thing I mentioned on the setter above: a very clear if / elif / else block that does repeat the get world poses call but is much cleaner overall.

Also the whole `if ..., raise ValueError` pattern should always just be an assertion.